Sources/ViewPosition.swift

```diff
@@ -0,0 +1,155 @@
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import CoreGraphics
+import UIKit
+
+
+public struct ViewPosition {
+    
+    // MARK: – Anchor
+    
+    public enum Anchor {
+        
+        case topLeft
+        case top
+        case topRight
+        
+        case left
+        case center
+        case right
+        
+        case bottomLeft
+        case bottom
+        case bottomRight
+        
+        public func anchorPoint(onRect rect: CGRect) -> CGPoint {
+            switch self {
+            case .topLeft:
+                return rect.origin
+            case .top:
+                return CGPoint(x: rect.maxX / 2.0, y: rect.minY)
+            case .topRight:
+                return CGPoint(x: rect.maxX, y: rect.minY)
+            case .left:
+                return CGPoint(x: rect.minX, y: rect.maxY / 2.0)
+            case .center:
+                return CGPoint(x: rect.maxX / 2.0, y: rect.maxY / 2.0)
+            case .right:
+                return CGPoint(x: rect.maxX, y: rect.maxY / 2.0)
+            case .bottomLeft:
+                return CGPoint(x: rect.minX, y: rect.maxY)
+            case .bottom:
+                return CGPoint(x: rect.maxX / 2.0, y: rect.maxY)
+            case .bottomRight:
+                return CGPoint(x: rect.maxX, y: rect.maxY)
+            }
+        }
+    }
+    
+    // MARK: – OffsetAnchor
+    
+    public struct OffsetAnchor {
+        
+        // MARK: Initialization
+        
+        public init(offset: UIOffset, anchor: Anchor) {
+            self.offset = offset
+            self.anchor = anchor
+        }
+        
+        // MARK: Internal Properties
+        
+        internal var offset: UIOffset
+        internal var anchor: Anchor
+        
+    }
+    
+    // MARK: Initialization
+    
+    public init(view: UIView, anchor: Anchor) {
+        self = ViewPosition(view: view, position: anchor.anchorPoint(onRect: view.bounds))
+    }
+    
+    public init(label: UILabel, anchor: Anchor) {
+        self = ViewPosition(view: label, position: anchor.anchorPoint(onRect: FontMetrics(for: label.font).textFrame(within: label.bounds)))
+    }
+    
+    public init(view: UIView, position: CGPoint) {
+        self.view = view
+        self.anchor = PixelRounder(for: view).roundToPixel(position)
+        originalBounds = view.bounds
+    }
+    
+    // MARK: Public Methods
+    
+    /// Aligns the receiver's view to the passed in ViewPosition.
+    public func align(to otherViewPosition: ViewPosition, xOffset: CGFloat = 0.0, yOffset: CGFloat = 0.0) {
+        guard view.bounds == originalBounds else {
+            assertionFailure("Bounds of view to align have changed since ViewPosition was created! \(originalBounds) -> \(view.bounds)")
```

Comment:
Per my other comment, tracking the position enum instead of a CGPoint will make this assertion unnecessary.